account_cancel_with_reversal/tests/test_account_move_reverse.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from openerp.tests import common
+
+
+class TestAccountInvoiceCancel(common.TransactionCase):
+    """
+    Case1 - test_invoice_cancel:
+    - Create draft invoice and validate and Cancel
+    - When invoice cancel reverse entry should be generated.
+    Case2 - test_voucher_cancel:
+    - Create draft voucher and validate and Unreconcile
+    - When voucher Unreconcile reverse entry should be generated.
+    """
+    def setUp(self):
+        super(TestAccountInvoiceCancel, self).setUp()
+        cr, uid = self.cr, self.uid
```

Comment:
I think you don't need to get cr, uid anymore for V8.
But you have to make sure you setup object by self.env['account.invoice']

Also, in this project, we will use Camel case for object, i.e., instead of 
`account_invoice_model = self.env['account.invoice']`
we should
`AccountInvoice = self.env['account.invoice']`

For this camel case, please just try to use it else where for good practice. Not that serious.